testsuite/tests/typing-layouts/modules.ml

```diff
@@ -340,10 +340,11 @@ type ('a : value) t4_val
 type t4 = M4.s t4_val;;
 [%%expect {|
 module F4 : functor (X : sig type t end) -> sig type s = Foo of X.t end
-module M4 : sig type s end
+module M4 : sig type s : immutable_data end
 type 'a t4_val
 type t4 = M4.s t4_val
 |}]
+(* CR layouts v2.8: The appearance of [immutable_data] there is just wrong. *)
```

Comment:
I'm pleasantly surprised by this output - I think this is right? In `F4`, `s` should have kind `immutable_data with X.t`. And then after the functor application, `X.t` is `t = T`, which is immediate. So `M4.s` really is `immutable_data`